src/fetch/PageAction.js

```diff
@@ -0,0 +1,113 @@
+import { pageAction } from "./prompts.js";
+import { logger } from "../log/logger.js";
+import { TagRemovingMinimizer } from "../min/TagRemovingMinimizer.js";
+import { getAI } from '../ai/index.js';
+import { Timer } from "../log/timer.js";
+
+export const PageAction = class {
+  constructor(namespace, prompt) {
+    this.namespace = namespace;
+    this.prompt = prompt;
+    this.ai = getAI()
+  }
+
+  async learn(examples) {
```

Comment:
- we should have a learn function which takes instance(s) of Document
- learn should be called once
- the idea is to learn how to do the action based on one example (or a handful of examples), and then not make AI calls in execute